### DISCUSSION TITLE
How to configure release name prefixes in monorepos?

### DISCUSSION THREAD

**Discussion by HenrikPoulsen** [renovatebot/renovate#19008] [Request Help] [answered]
I have an issue where Renovate tries to create the following PR for a go module:
`Update module cloud.google.com/go/iam to v0.107.0`
The iam package lives in this monorepo https://github.com/googleapis/google-cloud-go and when they ship a new version of iam the tag on github is `iam/{version}` as seen here https://github.com/googleapis/google-cloud-go/releases/tag/iam%2Fv0.7.0, but the version when using it in go is just `v0.7.0`.
The current latest version of iam is `v0.7.0` (as seen [here](https://pkg.go.dev/cloud.google.com/go/iam)) but Renovate is finding `v0.107.0` incorrectly, which is just the highest release version in their github releases list.
So what happens is that Renovate prints this error in the PR:
```
Command failed: docker run --rm --name=renovate_go --label=renovate_child {ommitted some vars}  -w "/tmp/renovate/repos/github/org/renovate-services" docker.io/renovate/go:1.19.3 bash -l -c "go get -d -t ./..."
go: cloud.google.com/go/iam@v0.107.0: reading cloud.google.com/go/iam/go.mod at revision iam/v0.107.0: unknown revision iam/v0.107.0
```

Is there some way to configure renovate to handle this?

Edit:
Clarified that the tag Renovate should look for is `iam/*`

**Comment by rarkins** [comment, marked as answer]
Unfortunately this is a known issue/bug. I'm pretty sure there's at least a couple of open bug issues right now which apply to it.

It seems the problem occurs when a "pseudo" name of sorts is used (e.g. cloud.google.com) which then points to github.com. In such cases the "iam" part from the original URL is lost by the time we're looking up on github.com

**Reply by HenrikPoulsen** [reply]
I tried playing around using packageRules. Like this for example:
```
{
    "matchPackageNames": ["cloud.google.com/go/iam"],
    "extractVersion": "^iam/(?<version>.*)$"
}
```
which doesn't work. Did some debugging and I see the getReleases() for golang basically throws out all the prefixed releases in the affected repo.
They get thrown away here 
https://github.com/renovatebot/renovate/blob/7d56162c492503b4da9b5ef2c610b128feaab0cb/lib/modules/datasource/go/releases-direct.ts#L128-L132
Changing that to for example this:
```js
if (res.releases) {
      res.releases = res.releases.filter((release) =>
        release.version?.split('/')?.at(-1)?.startsWith('v')
      );
}
```
allows those versions to be returned, which then lets the `extractVersion` function as I would expect.
Does that type of change make sense at all?
The code in there today already mistakenly includes non-version releases like [video/v1.10.0](https://github.com/googleapis/google-cloud-go/releases/tag/video%2Fv1.10.0) since they start with a v, so the code likely does need some tweaking it seems since I don't think that was intended.

**Reply by rarkins** [reply]
Yeah, we need to recursively try the whole git module path as prefix until we hit versions.
